---
title: "Introduction to RNA-seq"
teaching: 10
exercises: 0
questions:
- "What is RNA-seq?"
- "Sample preparation"
- "Experimental Design"
- "RNA library construction"
- "RNA-Seq analysis : Basic workflow"
objectives:
- "Understanding RNA-seq analysis"

keypoints:
- XXX
- YYY
- ZZZ
---

#### RNA-seq
- RNA sequencing (RNA-seq) is a revolutionary method used to study the transcriptome. 
- It is a highly sensitive and accurate tool for for simultaneously measuring and comparing the expression of tens of thousands of genes in multiple samples.
- It allows the researcher to interrogate the transcriptome under different disease states, in response to therapeutics, under different environmental conditions etc.
- As sequencing costs have decreased, using RNA-Seq to simultaneously measure the expression of tens of thousands of genes for multiple samples has never been easier.
&nbsp;

#### Steps involved in RNA sequencing
- There are multiple steps involved in an RNA-Seq experiment.
- The first is to identify the biological question which can be interrogated using RNA-Seq analysis.  
- Identify the experiment design and the underlying conditions of interest e.g. 'Treated' versus 'Untreated' samples.
- Isolate the RNA and prepare the RNA library using either single-end or paired-end reads.
- Sequencing the samples.

<p align="center">
  <img src="{{ page.root }}/fig/sample_to_sequencer.png" style="margin:10px;height:300px"/>
</p>


#### The Experimental Design
- Designing a RNA-Seq experiment is extremely crucial to the analysis.
- A **design** provides statistically sound results and  can provide answers to the experimental questions.
- **Replicates**   : Technical vs Biological.
- **Data amount/type** : Read length, single vs paired end, 
                         stranded vs unstranded, desired depth of coverage.
- **Recommended reads** :<br>
  * How many biological replicates are needed in an RNA-seq experiment and which differential expression tool should you use? [[Pubmed](https://pubmed.ncbi.nlm.nih.gov/27022035/)] <br>
  * Statistical design and analysis of RNA sequencing data [[Pubmed](https://pubmed.ncbi.nlm.nih.gov/20439781/)] <br>
  * Differentially expressed genes from RNA-Seq and functional enrichment results are affected by the choice of single-end versus paired-end reads and stranded versus non-stranded protocols [[Pubmed](https://pubmed.ncbi.nlm.nih.gov/28535780/)]

&nbsp;
#### Isolate the RNA
- RNA  has to be of a high quality (RNA integrity number). 
- Quality can be measured by a Agilent Bioanalyzer using a RNA integrity number (RIN). 
- The RIN can range from 1-10, 1 being worst, 10 being the best. 
&nbsp;
#### NGS library construction 
<p align="center">
<img src="{{ page.root }}/fig/chemistry.png" style="margin:10px;height:350px"/>
</p>

- Either you or your sequencing company would prepare libraries for sequencing. 
- Sequencing on Illumina platform is based on sequencing by synthesis technology where cDNA will bind to a flowcell, and is PCR amplified using bridge amplification to generate clusters on the flowcell. 
- Fluorescent signals are emitted with each incorporation of a nucleotide and the PCR ensures that there is enough fluorescent signal to be detected. There are a couple technical errors which can arise here â€“ main ones being sequencing errors and PCR artifacts
- Library contruction involves rRNA depletion (~90% of total RNA) or polyA enrichment to capture only mRNA.
- The next step is converting RNA to cDNA , fragmenting cDNA, and attaching adapters necessary for sequencing to work.
- RNA is then fragmented and reverse transcribed to cDNA (as the platform only sequences DNA) using random primers. Adapters are then ligated to the ends of the cDNA
- These libraries will attach to lanes on a flowcell through those adaptors and as they are being sequenced, a fluorescent light will be emitted, indicating to the machine what nucleotide was added.
- By the end, you will get single reads or paired end reads. Paired end reads are where the ends of a large fragment are sequenced and help to cover regions with low complexity (repetitive regions). It is also preferred if you are doing a de novo assembly of your reads. 
- The output from a sequencer is the sequence data with their quality scores in a FASTQ file format.

&nbsp;
#### Stranded vs Unstranded protocols
<p align="center">
<img src="{{ page.root }}/fig/Strand-Specific-Protocols-2.png" style="margin:10px;height:350px"/>
</p> 

- There are two types of RNA sequencing sample preparation protocols: [stranded and unstranded](https://www.ecseq.com/support/ngs/how-do-strand-specific-sequencing-protocols-work). 
- It is important to know which protocol was used in your experiment for your downstream analysis.
- Stranded protocols retain strandedness information (whether your RNA was transcribed from the forward or reverse strand). 
- Unstranded protocols do not retain this information.
- We will do a quick analysis to identify the strandedness 




{% include links.md %}
