---
title: "Two-step RNA-Seq analysis"
teaching: 10
exercises: 0
questions:
- "How to convert sequencing data to gene counts?"
- "How to identify differentially expressed (DE) genes and enriched functions?"

objectives:
- "Introduction to a two-step RNA-Seq analysis"
keypoints:
- We will use Nextflow pipeline nfcore-rnaseq for converting sequencing data to gene counts.
- We will use R/RStudio envirionment to identify differentially expressed (DE) genes and enriched functions.
---


- RNA-seq analysis involves simultaneous measurement and comparison of the expression of tens of thousands of genes in multiple samples.
- There are multiple steps involved in analysing RNA-Seq data.
- In addition many tools have been developed as alternative options at each step in the analysis.
- In this workshop we demonstrate how we can collapse all the steps into two distict workflows. 
- We will use two independant processing environments to excecute each part of the analysis.


### RNA-Seq analysis workflow 
<p align="center">
  <img src="{{ page.root }}/fig/original_pipe_to_nextflowing.png" style="margin:10px"/>
</p>

- **Part I: RNA-seq reads to counts (Using a nextflow pipeline)**
  * The analysis begins with sequencing reads (FASTQ files).
  * Perform quality control of the reads using tools such as [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and [Trim galore](https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/).
  * The good quality reads are then usually aligned to a reference genome using tools such as [STAR](https://github.com/alexdobin/STAR) or [HiSAT2](http://daehwankimlab.github.io/hisat2/#:~:text=HISAT2%20is%20a%20fast%20and,to%20a%20single%20reference%20genome.). 
  * The number of reads mapped to each gene can then be counted. This results in a table of read-counts per gene.
  * The nextflow pipeline [nfcore-rnaseq](https://nf-co.re/rnaseq/3.8.1) can be used to run part I  with a single command.
  
<br><br>  
- **Part II: RNA-seq counts to genes and functional enrichments (Using RStudio IDE)** 
  * The gene-counts can be used to perform statistical analyses to determine differentially expressed (DE) genes and pathways.
  * Multiple independant packages/libraries developed in [R-programming](https://www.r-project.org/) can be used for this part of the analysis. The [RStudio IDE](https://www.rstudio.com/) can be used to integrate the R-code. 
  * R packages such as [DeSeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html), [EdgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) etc are used for identification of differencially expressed (DE) genes.
  * R packages such as [clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html) and [pathfindR](https://cran.r-project.org/web/packages/pathfindR/index.html) and many others have been developed for functional enrichment analysis. 
  * In addition to R libraries, independant web-tools such as [PANTHER](http://www.pantherdb.org/), [DAVID](https://david.ncifcrf.gov/) and commercial tools such as [Ingenuity Pathway Analysis (IPA)](https://digitalinsights.qiagen.com/products-overview/discovery-insights-portfolio/analysis-and-visualization/qiagen-ipa/?cmpid=QDI_GA_IPA&gclid=Cj0KCQjwkruVBhCHARIsACVIiOxh5HnHmJy72fRrmfQlnIhUpy2mpaFuVE2ClZDctskvzCoU-XCQerMaAmOhEALw_wcB) and [Metacore](https://clarivate.com/cortellis/webinars/metacore-integrated-pathway-analysis-for-multi-omics-data/) are also available.

