---
title: "Run nfcore-rnaseq on a Nimbus instance"
teaching: 10
exercises: 0
questions:
- "How to provide sequence data input to nfcore-rnaseq pipeline?"
- "How to identify strandedness of the data?"
- "How to provide reference data to nfcore-rnaseq pipeline?"
- "How to assign cpu/memory resources to the Nimbus VM instance?"
- "How to assign the software tools to nfcore-rnaseq pipeline?"
- "What does the basic command of nfcore-rnaseq pipeline look like?"

objectives:
- "Convert the read-sequencing data into gene counts"

keypoints:
- nfcore-rnaseq pipeline includes many alternative tools at multiple steps of the rnaseq analysis process.
- Containers such as singularity and docker are integrated in nextflow environment.

---

<figure>
  <img src="{{ page.root }}/fig/STEP1_original_pipe_to_nextflowing.png" style="margin:10px;height:400px"/>
  <figcaption> RNA-Seq analysis workflow </figcaption>
</figure><br>


### Samplesheet for sequence data
- You will need a [samplesheet](https://nf-co.re/rnaseq/3.7/usage#samplesheet-input) with information about the samples you would like to analyse. 
- It has to be a comma-separated file with 4 columns, and a header row as shown in the example samplesheet below. 
- If the samplesheet is located in a different folder, you can use this parameter to specify its location. 

<figure>
  <img src="{{ page.root }}/fig/elaborate_samplesheet.png" style="margin:10px;height:150px"/>
</figure><br>

- The four columns in the samplesheet are as follows:
<figure>
  <img src="{{ page.root }}/fig/samplesheet_description.png" style="margin:10px;height:200px"/>
</figure><br>

- The samplesheet required for today's analysis is a lot simpler, as we are using single-end reads. 
- This samplesheet is already created and placed in the folder XXX.
- However one thing we can check ourselves as a task, is the **strandedness** of the read library.

> ### Identify strandedness
> - Are the reads in our dataset stranded or unstranded?
> - As far as I know, this data is unstranded, but as a sanity check we can confirm. 
> - You can use RSeQC Infer Experiment tool to “guess” the strandness, as explained in the RNA-seq ref-based tutorial.
> - 
> ```
> COMMAND for RSeQC
> 
> ```

> {: .language-bash}
{: .solution}


### Species references
```
- Details to be provided here 
- Either locally placed 
- Or CernVM-FS
```

### Quick Start
```
nextflow run nf-core/rnaseq \
    --input samplesheet.csv \
    --outdir <OUTDIR> --genome GRCh38 
    -profile <docker/singularity/podman/shifter/charliecloud/conda/institute>
```

- Note that some kind of configuration will be needed so that Nextflow knows how to fetch the required software. This is usually done in the form of a config profile. 
- The pipeline comes with config profiles such as docker, singularity, podman, shifter, charliecloud and conda which instruct the pipeline to use the named tool for software management.
- Please check [nf-core/configs](https://github.com/nf-core/configs#documentation) to see if a custom config file to run nf-core pipelines already exists for your Institute. If so, you can simply use -profile <institute> in your command. This will enable either docker or singularity and set the appropriate execution settings for your local compute environment.

### Software profiles
- We can select a specific profile such as **singularity**, and use the [nf-core](https://nf-co.re/tools/#downloading-pipelines-for-offline-use) download command to download images first, before running the pipeline. 
- Nextflow allows us to store and re-use the images from a central location for future pipeline runs.
  







  

