---
title: "The nfcore-rnaseq command"
teaching: 10
exercises: 0
questions:
- "What are the main input parameters?"
- "What are the main log files?"
- "What are the main outputs?"

objectives:
- "Understanding nfcore-rnaseq inputs and outputs"

keypoints:
- Reference daatsets can be pre-downloaded or downloaded/indexed on the run.
- System resouces such as cpus and memory can be customised per step using config files.
- Excecution and timeline logs files provide a detailed summary of the run.
---


### nfcore-rnaseq command

~~~
base_path='DEFINE BASE PATH'

local_rnaseq_path='PATH_TO_DOWNLOADED_nfcore-rnaseq_IMAGE'

nextflow run $local_rnaseq_path/rnaseq-3.7 \
        --input samplesheet.csv \                                                      # samplesheet file-name
        --outdir results \                                                             # Define the name of the results folder
        --genome GRCm38 \                                                              # Reference genome
        -profile singularity \                                                         # profile e.g. singularity
        --fasta $base_path/Mouse_genomeFiles_from_sarek/genome.fa \                    # Path to the downloaded genome fasta file
        --gtf $base_path/Mouse_genomeFiles_from_sarek/sarek_Mmus.gtf \                 # Path to the downloaded 'gtf' file 
        --max_memory '64 GB' --max_cpus 2 \                                            # Define resources          
        --star_index $base_path/Mouse_genomeFiles_from_sarek/STARIndex_sarekGenome/ \  # Path to the downloaded genome 'STAR' mapper index file
        -with-report excecution_report_maxcpu2.html \                                  # Excecution log file-name (generated by nextflow)
        -with-timeline timeline_report_maxcpu2.html \                                  # Timeline log file-name.  (generated by nextflow)
        -r 3.4                        
~~~



  
