---
title: "RNA-Seq counts to Differentially Expressed genes"
teaching: 10
exercises: 0
questions:
- "What is the output of nfcore-rnaseq?"
- "What is the next step?"
- "What environment do we use for downstream analysis? - R/RStudio"

objectives:
- "Discuss the step 2 in RNA-Seq analysis using RStudio"

keypoints:
-

---

<figure>
  <img src="{{ page.root }}/fig/STEP2_original_pipe_to_nextflowing.png" style="margin:10px;height:400px"/>
  <figcaption> RNA-Seq analysis workflow </figcaption>
</figure><br>


### RNA-seq counts to genes and functional enrichments (Using RStudio IDE) 
- The gene-counts can be used to perform statistical analyses to determine differentially expressed (DE) genes and pathways.
- Multiple independant packages/libraries developed in [R-programming](https://www.r-project.org/) can be used for this part of the analysis. The [RStudio IDE](https://www.rstudio.com/) can be used to integrate the R-code. 
- R packages such as [DeSeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html), [EdgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) etc are used for identification of differencially expressed (DE) genes.
- R packages such as [clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html) and [pathfindR](https://cran.r-project.org/web/packages/pathfindR/index.html) and many others have been developed for functional enrichment analysis. 
- In addition to R libraries, independant web-tools such as [PANTHER](http://www.pantherdb.org/), [DAVID](https://david.ncifcrf.gov/) and commercial tools such as [Ingenuity Pathway Analysis (IPA)](https://digitalinsights.qiagen.com/products-overview/discovery-insights-portfolio/analysis-and-visualization/qiagen-ipa/?cmpid=QDI_GA_IPA&gclid=Cj0KCQjwkruVBhCHARIsACVIiOxh5HnHmJy72fRrmfQlnIhUpy2mpaFuVE2ClZDctskvzCoU-XCQerMaAmOhEALw_wcB) and [Metacore](https://clarivate.com/cortellis/webinars/metacore-integrated-pathway-analysis-for-multi-omics-data/) are also available.
